server/planning/planning/planning_featured_async_service.py

```diff
@@ -0,0 +1,113 @@
+from datetime import datetime
+from typing import Any
+from copy import deepcopy
+
+from apps.archive.common import update_dates_for
+from apps.auth import get_user_id
+from planning.core.service import BasePlanningAsyncService
+from planning.common import (
+    get_version_item_for_post,
+    enqueue_planning_item,
+    set_original_creator,
+)
+from planning.planning.planning_service import PlanningAsyncService
+from planning.types import PlanningFeaturedResourceModel
+from superdesk import get_resource_service, logger
+from superdesk.core import get_app_config
+from superdesk.core.types import SearchRequest
+from superdesk.errors import SuperdeskApiError
+from superdesk.utc import utc_to_local, utcnow
+
+ID_DATE_FORMAT = "%Y%m%d"
+
+
+class PlanningFeaturedAsyncService(BasePlanningAsyncService[PlanningFeaturedResourceModel]):
+    """Async Service class for the planning featured model."""
+
+    async def on_create(self, docs: list[PlanningFeaturedResourceModel]):
+        for doc in docs:
+            doc_dict = doc.to_dict()
+            date = utc_to_local(doc_dict.get("tz") or get_app_config("DEFAULT_TIMEZONE"), doc_dict.get("date"))
+            _id = date.strftime(ID_DATE_FORMAT)
+
+            search_request = SearchRequest(where={"_id": _id})
+            items = await super().find(search_request)
+            if await items.count() > 0:
+                raise SuperdeskApiError.badRequestError(message="Featured story already exists for this date.")
+
+            await self.validate_featured_attrribute(doc_dict.get("items", []))
+            doc_dict["_id"] = _id
+            await self.post_featured_planning(doc_dict)
+            # set the author
+            set_original_creator(doc_dict)
+
+            # set timestamps
+            update_dates_for(doc_dict)
+            doc = PlanningFeaturedResourceModel(**doc_dict)
```

Comment:
If you update this service to use a ResourceModel instance, you can then update the instance values in other functions, rather than doing this (re-assigning while looping is also not ideal)